modules/tinymce/src/plugins/fullscreen/main/ts/core/Actions.ts

```diff
@@ -105,17 +177,29 @@ const toggleFullscreen = (editor: Editor, fullscreenState: Cell<any>) => {
     DOM.removeClass(editorContainer, 'tox-fullscreen');
 
     viewportUpdate.unbind();
+    Optional.from(fullscreenState.get()).each((info) => info.fullscreenChangeHandler.unbind());
```

Comment:
Why not just make fullscreenState either a `Cell<Optional<ScrollInfo>>` or a `Singleton.Value<ScrollInfo>`?

Reply:
I did originally start converting to Singleton.Value but then I realized that it doesn't really have a good way of dealing with the toggling. Essentially for the toggle you have to do:
```
if (!singleton.isSet()) {
  singleton.set({
    // ...
  });
} else {
  singleton.on((value) => {
    // ....
   singleton.clear();
  });
}
```
So I decided that the code was actually clearer as it was.

Reply:
One way we could do it, is if we used an option instead of a singleton, we could break the enable/disable logic out into separate functions and then use fold to branch. That, however, is really outside the scope of this PR.